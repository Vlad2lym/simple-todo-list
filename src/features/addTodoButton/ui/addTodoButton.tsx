import addIcon from '@/shared/images/icons/add-icon.svg';
import styles from './addTodoButton.module.scss';

interface IProps {
  className?: string;
  onClick: () => void;
}

export const test_AddTodoBtn = 'test_AddTodoBtn';

export const AddTodoButton = ({ className, onClick }: IProps) => {
  return (
    <div className={className ? `${styles.addTodoArea}} ${className}` : styles.addTodoArea}>
      <button className={styles.addTodoButton} onClick={onClick} data-testid={test_AddTodoBtn}>
        <img src={addIcon} />
      </button>
    </div>
  );
};
