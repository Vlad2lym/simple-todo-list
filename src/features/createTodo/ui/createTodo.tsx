import { useCallback, useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { Todo } from '@/entities/todo';
import { TodoForm } from '@/entities/todoForm';

interface IProps {
  onApply: (todoInfo: Todo) => void;
  onCancel: () => void;
  order: number;
}

export const CreateTodoModal = ({ onApply, onCancel, order }: IProps) => {
  const [titleTodo, setTitleTodo] = useState('');

  const onChangeTitleTodo = (value: string) => {
    setTitleTodo(value);
  };

  const onFormFinish = useCallback(() => {
    if (!titleTodo.trim()) {
      return;
    }

    const todoInfo: Todo = {
      id: uuidv4(),
      title: titleTodo,
      active: true,
      order: order,
    };

    onApply(todoInfo);
    setTitleTodo('');
    onCancel();
  }, [titleTodo, onApply, onCancel, order]);

  return (
    <>
      <TodoForm
        title="NEW NOTE"
        onCancel={onCancel}
        onApply={onFormFinish}
        inputValue={titleTodo}
        onChangeInput={onChangeTitleTodo}
      />
    </>
  );
};
