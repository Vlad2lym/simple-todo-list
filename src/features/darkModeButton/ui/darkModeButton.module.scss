.dark-mode-button {
  width: 38px;
  height: 38px;
  background-color: var(--primary-color);
  background-image: var(--mode-button-icon);
  background-repeat: no-repeat;
  background-position: center;
  border: none;
  border-radius: 5px;
  cursor: pointer;

  &:hover {
    background-color: var(--primary-color-active);
    box-shadow: 0 0 4px 0 rgba(108, 99, 255, 1);
  }
}
