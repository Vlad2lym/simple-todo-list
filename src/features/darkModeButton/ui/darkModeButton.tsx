import styles from './darkModeButton.module.scss';

interface IProps {
  onChange: () => void;
}

export const test_DarkModeBtn = 'test_DarkModeBtn';

export const DarkModeButton = ({ onChange }: IProps) => {
  return <button className={styles.darkModeButton} onClick={() => onChange()} data-testid={test_DarkModeBtn} />;
};
