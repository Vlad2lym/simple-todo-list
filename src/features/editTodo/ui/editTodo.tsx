import { useCallback, useState } from 'react';
import { Todo } from '@/entities/todo';
import { TodoForm } from '@/entities/todoForm';

interface IProps {
  onApply: (todoInfo: Todo) => void;
  onCancel: () => void;
  editedTodo: Todo | null;
}

export const EditTodoModal = ({ onApply, onCancel, editedTodo }: IProps) => {
  const [titleTodo, setTitleTodo] = useState(editedTodo?.title ?? '');

  const onChangeTitleTodo = (value: string) => {
    setTitleTodo(value);
  };

  const onFormFinish = useCallback(() => {
    if (!titleTodo.trim() || !editedTodo) {
      return;
    }

    const todo: Todo = {
      id: editedTodo.id,
      title: titleTodo,
      active: editedTodo.active,
      order: editedTodo.order,
    };

    onApply(todo);
    onCancel();
  }, [titleTodo, onApply, onCancel, editedTodo]);

  return (
    <>
      <TodoForm
        title="EDIT NOTE"
        onCancel={onCancel}
        onApply={onFormFinish}
        inputValue={titleTodo}
        onChangeInput={onChangeTitleTodo}
      />
    </>
  );
};
