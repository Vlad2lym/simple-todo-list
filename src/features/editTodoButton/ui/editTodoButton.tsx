import styles from '../styles/editTodoButton.module.scss';

interface IProps {
  idTodo: string;
  onClick: (id: string) => void;
}

export const EditTodoButton = ({ idTodo, onClick }: IProps) => {
  return (
    <>
      <button className={styles.editBtn} onClick={() => onClick(idTodo)} />
    </>
  );
};
