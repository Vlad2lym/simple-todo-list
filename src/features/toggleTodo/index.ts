export { ToggleTodo } from './ui/toggleTodo';
