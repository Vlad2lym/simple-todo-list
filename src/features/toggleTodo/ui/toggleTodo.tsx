import { TodoInfo } from '../../../shared/api';
import { CustomCheckbox } from '../../../shared/ui/customCheckbox';

interface IProps {
  todo: TodoInfo;
  toggleTodoById: (id: string) => void;
}

export const ToggleTodo = ({ todo, toggleTodoById }: IProps) => {
  return <CustomCheckbox id={todo.id} checked={!todo.active} onChange={(id) => toggleTodoById(id)} />;
};
