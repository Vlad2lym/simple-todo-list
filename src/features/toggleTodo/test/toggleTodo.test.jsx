import { render, screen } from '@testing-library/react';
import { ToggleTodo } from '../ui/toggleTodo';
import { jest } from '@jest/globals';
import userEvent from '@testing-library/user-event';

const todo = {
  id: '1',
  title: 'title1',
  active: true,
  order: 1,
};

const todoCompleted = {
  id: '1',
  title: 'title1',
  active: false,
  order: 1,
};

const onClick = jest.fn();

it('should toggle todo on click', async () => {
  const { rerender, container } = render(<ToggleTodo todo={todo} toggleTodoById={onClick} />);
  const checkbox = screen.getByRole('checkbox');
  expect(container.firstChild).toMatchSnapshot();
  expect(onClick).not.toHaveBeenCalled();
  await userEvent.click(checkbox);
  expect(onClick).toHaveBeenCalledTimes(1);
  expect(onClick).toHaveBeenCalledWith('1');
  expect(screen.getByRole('checkbox', { checked: false })).toBeInTheDocument();
  rerender(<ToggleTodo todo={todoCompleted} toggleTodoById={onClick} />);
  expect(screen.getByRole('checkbox', { checked: true })).toBeInTheDocument();
});
