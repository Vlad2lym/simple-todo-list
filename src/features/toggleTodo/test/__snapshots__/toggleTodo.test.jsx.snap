// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should toggle todo on click 1`] = `
<div
  class="wrapper"
>
  <input
    class="checkbox"
    id="1"
    type="checkbox"
  />
  <label
    for="1"
  />
</div>
`;
